domain/src/main/java/org/oppia/domain/exploration/ExplorationDataController.kt

```diff
@@ -0,0 +1,289 @@
+            /* keyName= */"x", interactionId))
+          .build())
+    }
+    return ruleSpecList
+  }
+
+  // Creates an input interaction object from JSON
+  private fun createInputFromJson(inputJson: JSONObject?,
+                                              keyName: String,
+                                              interactionId: String): InteractionObject {
+    if(inputJson == null) {
+      return InteractionObject.getDefaultInstance()
+    }
+    if(interactionId == "MultipleChoiceInput") {
+      return InteractionObject.newBuilder()
+        .setNonNegativeInt(inputJson.getInt(keyName))
+        .build()
+    } else if (interactionId == "TextInput") {
+      return InteractionObject.newBuilder()
+        .setNormalizedString(inputJson.getString(keyName))
+        .build()
+    } else if (interactionId == "NumericInput") {
+      return InteractionObject.newBuilder()
+        .setReal(inputJson.getDouble(keyName))
+        .build()
+    } else {
+      return InteractionObject.getDefaultInstance()
+    }
+  }
+
+  // Creates a customization arg mapping from JSON
+  private fun createCustomizationArgsMapFromJson(
+    customizationArgsJson: JSONObject?): MutableMap<String, InteractionObject> {
+    val customizationArgsMap: MutableMap<String, InteractionObject> = mutableMapOf()
+    if (customizationArgsJson == null) {
+      return customizationArgsMap
+    }
+    val customizationArgsKeys = customizationArgsJson.keys()?: return customizationArgsMap
+    val customizationArgsIterator = customizationArgsKeys.iterator()
+    while(customizationArgsIterator.hasNext()) {
+      val key = customizationArgsIterator.next()
+      customizationArgsMap[key] =  createCustomizationArgValueFromJson(
+        customizationArgsJson.getJSONObject(key).get("value"))
+    }
+    return customizationArgsMap
+  }
+
+  // Creates a customization arg value interaction object from JSON
+  private fun createCustomizationArgValueFromJson(customizationArgValue: Any): InteractionObject {
+    val interactionObjectBuilder = InteractionObject.newBuilder()
+    when(customizationArgValue) {
+      is String -> return interactionObjectBuilder
+        .setNormalizedString(customizationArgValue).build()
+      is Int -> return interactionObjectBuilder
+        .setSignedInt(customizationArgValue).build()
+      is Double -> return interactionObjectBuilder
+        .setReal(customizationArgValue).build()
+      is List<*> -> if(customizationArgValue.size > 0) {
+        return interactionObjectBuilder.setSetOfHtmlString(
+          createStringList(customizationArgValue)).build()
+      }
+    }
+      return InteractionObject.getDefaultInstance()
+  }
+
+  @Suppress("UNCHECKED_CAST") // Checked cast in the if statement
+  private fun createStringList(value: List<*>): StringList {
+    val stringList = mutableListOf<String>()
+    if(value[0] is String) {
+      stringList.addAll(value as List<String>)
+    }
+    return StringList.getDefaultInstance()
```

Comment:
Should we be returning stringList in the if block?

Reply:
Done